File: Backjoon problems Java/Baekjoon_14916_1_Gkuer.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.lang.reflect.Array;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        int inf = 987654321;
        int[] temps = {0, inf, 1, inf, 2, 1};
        int[] maps = new int[n+1];

        if (n <= 5) {
            for (int i = 1; i < n+1; i++) {
                maps[i] = temps[i];
            }
        } else {
            for (int i = 1; i < 6; i++) {
                maps[i] = temps[i];
            }
        }

        for (int i = 6; i < n+1; i++) {
            maps[i] = Math.min(maps[i - 2], maps[i - 5])+1;
        }

        if (maps[n] >= inf) {
            bw.write(-1 + "");
        } else {
            bw.write(maps[n] +"");
        }

        bw.flush();
    }
}

// Pass

File: Baekjoon problems Python 2/Baekjoon_14916_1_Gkuer.py
inf = 987654321
n = int(input())
maps = [inf for _ in range(n+1)]
temps = [0, inf, 1, inf, 2, 1]

if n >= 5:
    maps[1], maps[2], maps[3], maps[4], maps[5] = inf, 1, inf, 2, 1
else:
    for i in range(1,n+1):
        maps[i] = temps[i]

for i in range(6, n+1):
    maps[i] = min(maps[i-2],maps[i-5])+1

if maps[n] >= inf:
    print(-1)
else:
    print(maps[n])

# Pass